addons/medical_treatment/functions/fnc_scanMedicalItems.sqf

```diff
@@ -0,0 +1,34 @@
+#include "script_component.hpp"
+/*
+ * Author: Salluci
+ * Caches all item classnames used in ACE_Medical_Treatment_Actions
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * call ace_medical_treatment_fnc_scanMedicalItems
+ *
+ * Public: No
+ */
+
+private _list = [];
+private _cfgActions = configFile >> QGVAR(actions);
+
+private _fnc_isMedicalItem = toString {
+    getNumber (_x >> "ACE_isMedicalItem") isEqualTo 1
+};
+
+// get items in ACE_Medical_Treament_Actions, fallback for items without API config property
+{
+    _list append (getArray (_x >> "items"));
+} forEach ("true" configClasses _cfgActions);
+
+{
+    _list pushBack (configName _x);
+} forEach (_fnc_isMedicalItem configClasses (configFile >> "CfgWeapons"));
+
+uiNamespace setVariable [QGVAR(treatmentItems), compileFinal str (_list arrayIntersect _list)]
```

Comment:
Why are you storing code? I don't like this, this should store an array that gets cache-invalidated as necessary. Not sure how other things in Arsenal are done though.

Reply:
Do we do that for other variables in `ace_arsenal`?

Reply:
Nope.

Reply:
Then I really don't see the reason. @PabstMirror ?

Reply:
this is how we've been storing things computed at prestart
see https://github.com/acemod/ACE3/blob/master/addons/cargo/XEH_preStart.sqf

Reply:
I can swing either way, but arsenal just leaves things unprotected (performance concerns, I guess). I guess once 2.14 compileFinal hits this can be switch to a proper data type anyway.